Sources/GroundState/py_variational_montecarlo.hpp

```diff
@@ -160,15 +160,198 @@ void AddVariationalMonteCarloModule(py::module &m) {
       .def(
           "get_observable_stats",
           [](VariationalMonteCarlo &self) {
-            py::dict data;
             self.ComputeObservables();
-            self.GetObsManager().InsertAllStats(data);
-            return data;
+            return self.GetObservableStats();
           },
           R"EOF(
         Calculate and return the value of the operators stored as observables.
 
-        )EOF");
+        )EOF")
+      .def_property_readonly("vmc_data", &VariationalMonteCarlo::GetVmcData);
+
+  py::class_<vmc::Result>(m_vmc, "_VmcResult");
+
+  m_vmc.def("compute_samples", vmc::ComputeSamples, py::arg("sampler"),
+            py::arg("nsamples"), py::arg("ndiscard") = 0,
+            py::arg("compute_logderivs") = true, R"EOF(
+           Computes a sequence of visible configurations based on Monte Carlo sampling using `sampler`.
+
+           Args:
+               sampler: The sampler used to perform the MC sweeps.
+               nsamples: The number of MC samples that are stored (one MC sweep
+                    is performed between each sample)
+               ndiscard: The number of sweeps to be discarded before starting
+                    to store samples.
+               compute_logderivs: Whether to store the logarithmic derivatives
+                    of the wavefunction as part of the returned VMC result.
```

Comment:
what does it happen if `compute_logderivs=False` and one tries to compute the gradient using this `vmc_data`?

Reply:
It will throw an exception, like in this test:
https://github.com/netket/netket/blob/12b8e68abf931c9ef44d078ac9bbd924b769c293/Test/GroundState/test_vmc.py#L60-L68

Reply:
OK very good. I just have the feeling that by default this should be False and not True...

Reply:
I think the fact that most of our current code requires the log-derivatives suggests that computing them by default makes sense for us. Omitting them appears more like an optimization to me (which might be useful if your system is large and you really just need energy and variance).